src/utils.js

```diff
@@ -0,0 +1,51 @@
+// Recursively iterate over an evolution chain and normalize into an array.
+export const normalizeEvolutionChain = ( evolution ) => {
+	if ( evolution.length === 0 || evolution.evolves_to.length === 0 ) {
+		return [];
+	}
+
+	const triggersDisplayName = {
+		'level-up': 'Lvl',
+		trade: 'Trade',
+		'use-item': 'Use',
+	};
+
+	const nextEvolution = evolution.evolves_to[ 0 ],
+		details = nextEvolution.evolution_details[ 0 ],
+		currentId = extractId( evolution.species.url ),
+		nextId = extractId( nextEvolution.species.url );
+
+	const normalized = {
+		currentId,
+		nextId,
+		currentName: evolution.species.name,
+		nextName: nextEvolution.species.name,
+		currentImage: getImageURL( currentId ),
+		nextImage: getImageURL( nextId ),
+		trigger: triggersDisplayName[ details.trigger.name ],
+		triggerValue: details.min_level || details.min_happiness || details.item?.name.replace( '-', ' ' ) || '',
+	};
+
+	return [
+		normalized,
+		...normalizeEvolutionChain( nextEvolution ),
+	];
+};
+
+// Extract pokemon id from URL.
+export const extractId = ( url ) => {
+	return url.match( /\/(\d+)\// )[ 1 ];
```

Comment:
Consider adding the regex structure in the comment + are you sure you ill always have [ 1 ] in the match result?

Reply:
Moved